### PR TITLE
Refactor config to allow it to be overridden via env vars.

### PR DESCRIPTION
**Pull request by unscriptable**
This refactor moves the configuration out into it's own component folder while also allowing many config variables to be overridden by env vars.  See the README file and the `src/app/configure/` folder for more info.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,15 +15,32 @@ To run the web server:
...
+```bash
+EMAIL_PASS="really bad password" npm run gulp
+```
+
+See the `app/configure` files for more `env` var overrides, such as `PORT`:
+
+```bash
+PORT=8080 EMAIL_PASS="really bad password" npm run gulp
+```
+
```
Comment (kianoshp) [inline README.md:30]: So this is what we decided on? As an env variable? should we consider Firebase (https://firebase.google.com/) as an option?

File: README.md [hunk 2]
```diff
@@ -15,15 +15,32 @@ To run the web server:
...
+page, but the rest of the templates are rendered via a super simple template
+function that uses `${token}` place markers.  We should decide to use either
+React or the super simple template function.
+
+I refactored the site from a buckets-of-type structure to a component-oriented
+structure.  The view templates have been moved to the `src` folder.  The old
+templates in the `views` folder are no longer used.  
 
-I haven't touched the css, yet.
+I haven't moved the css into the component folders, yet.
```
Comment (kianoshp) [inline README.md:44]: Do I need to do this?

File: README.md (hunk 1 above)
Comment (unscriptable) [reply README.md:30]: imho, the env var is good enough for now.

File: README.md (hunk 2 above)
Comment (unscriptable) [reply README.md:44]: Not unless you want to.  I have no idea how much effort it will take.

**Pull request merged**